### ISSUE TITLE
About the pretrain on Waymo dataset.

### ISSUE THREAD

**Issue by jetpackfirstme** [CVMI-Lab/ST3D#41]
Hello, Jihan! First, thans for your excellent work! I am trying to reproduce the performance on Waymo->NuScenes. However I meet some questions. I wonder if you can give me some advice, thank you!

1. I run your codes based on OpenPCDet 0.5 and SPConv 1.2, and I made some modifications to fit OpenPCDet 0.5.
2. For the pretrainin stage, I use the config from ST3D/tools/cfgs/da-waymo-nus_models/secondiou/secondiou.yaml.
3. However, after I finish the pretrain, I test the final chekcpoint on NuScenes using the config from ST3D/tools/cfgs/da-waymo-nus_models/secondiou_st3d/secondiou_st3d.yaml, and the results are very weird:
![Screenshot from 2022-04-13 20-11-30](https://user-images.githubusercontent.com/22596812/163177366-6308cab6-e58a-400e-97d1-4fa91a39156a.png)
As you can see, bbox AP and bev AP both seems reasonable, however 3d AP is almost zero, I checked my codes and have no idea why this happens. Could you please give me some insights on this problem?
4. I also noticed that the UPDATE_PSEUDO_LABEL_INTERVAL is 1000 in config from ST3D/tools/cfgs/da-waymo-nus_models/secondiou_st3d/secondiou_st3d.yaml, which means you only generate pseudo label once at the begining of the self-train stage. Should I modify this?

It would be very kind of you to reply me! Thank you very much!

**Comment by jihanyang** [comment]
I want to hightlight that update ST3D to openpcdet latest version is not easy, since I have also failed. I guess this is caused by the error when update to openpcdet v0.3, which have been found before, it cannot even provide a correct evaluation result. We have an internal version which can work on openpcdet v0.3.1. Since there are many users report this problem and have the demand to work on more latest openpcdet version, I will release this internal version maybe this weekend. You only need to support spconv2.x in that version, which should be work normally.

**Comment by jihanyang** [comment]
Updating: I already modified ST3D to fit openpcdet v0.3. However, it cannot provide correct evaluation results for previous pretrained model. I need more time to further verify whether this is caused by different pytorch and spconv version during training and testing.

**Comment by jihanyang** [comment]
@jetpackfirstme 
For your confusion, could you visualize some frames? Maybe this is caused by error in z-axis. I find that in this task, updating pseudo label will cause performance degradation.
